### ISSUE TITLE
dateTimeFns.add() does not constrain when not presented with full precision of numbers

### ISSUE THREAD

**Issue by kolton-musgrove** [rubber-duck-software/iso-fns#64]
```
    it('constrain when not presented with full precision', () => {
      assert.equal(`${dateTimeFns.add(jan31, { months: 1 })}`, '2020-02-29T15:00:00')
      assert.equal(`${dateTimeFns.add(jan31, { months: 1 }, { overflow: 'constrain' })}`, '2020-02-29T15:00:00')
    })
```

Thie bug is occurring because the method of converting durationLike to duration creates slots even for those not present on the original durationLike object. This new set of slots are then added to the original date-time and are considered valid, real values.

**Comment by kolton-musgrove** [comment]
Same error exists with dateTimeFns.subtract().

**Comment by rubber-duck-software** [comment]
This is resolved in 2.0.0-alpha.8
